go/cmd/gno_social_feed_indexer/clientql/client.go

```diff
@@ -0,0 +1,120 @@
+package clientql
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/TERITORI/teritori-dapp/go/internal/indexerdb"
+	indexerql "github.com/TERITORI/teritori-dapp/go/pkg/indexerQL"
+	"github.com/TERITORI/teritori-dapp/go/pkg/networks"
+	"github.com/gofrs/uuid"
+	"gorm.io/gorm"
+)
+
+type IndexerQL struct {
+	gqlClient graphql.Client
+	db        *gorm.DB
+	networkID string
+}
+
+func New(networkID string, graphqlEndpoint string, db *gorm.DB) *IndexerQL {
+	gqlClient := graphql.NewClient(graphqlEndpoint, nil)
+	return &IndexerQL{gqlClient: gqlClient, db: db, networkID: networkID}
+}
+
+func (client *IndexerQL) SyncPosts() error {
+	lastPost := client.getLastPost()
+	var fromBlock int
+	if lastPost != nil {
+		fromBlock = int(lastPost.CreatedAt) + 1
+	}
+
+	posts, err := indexerql.GetPostTransactions(context.Background(), client.gqlClient, fromBlock)
+	if err != nil {
+		return err
+	}
+
+	for _, transaction := range posts.Transactions {
+		for _, post := range transaction.Messages {
+			data, ok := post.Value.(*indexerql.GetPostTransactionsTransactionsTransactionMessagesTransactionMessageValueMsgCall)
+			if !ok {
+				fmt.Println("not ok, skipping")
```

Comment:
the policy on current indexers is to stop in unexpected cases

also use a zap logger instead of fmt please

Reply:
Added zap logger and for now keeping it as it was